### ISSUE TITLE
Running Grobid as rootless container in Kubernetes

### ISSUE THREAD

**Issue by vipulg13** [kermitt2/grobid#1141]
Dear team,

currently, the Grobid service requires root privileges to be running in a Kubernetes environment. However, running a container as root in a production environment is not recommended. Are there any plans to have Grobid service running as rootless container or nonroot privileged container (with a specific id)?

E.g.,: The container fails in the intialisation process when run as nonroot due to issues creating tmp directory and read-only file system:

![image](https://github.com/kermitt2/grobid/assets/74742478/9c63723d-b45d-43dd-b9c2-76f32ee9147a)

**Comment by lfoppiano** [comment]
Thanks for reporting this issue @vipulg13. I'm not sure how long may take, it could be to re-engineer the whole build. 
We can find a solution on that for version 0.8.2 as we are freezing changes for version 0.8.1.

**Comment by qhaas** [comment]
This would be useful for docker compose users as well.  By default, it seems to run as root.

**Comment by lfoppiano** [comment]
I agree, unfortunately I'm too busy with other more pressing tasks and I cannot dedicate time to this. I'm happy to revise and integrate proposals that could solve this issue. 🙂

**Comment by lfoppiano** [comment]
Grobid does not write any file, so it should be easy to run it without root:

```
docker run  --user $(id -u):$(id -g) --rm --init --ulimit core=0 -p 8080:8070 -p 8081:8071 grobid/grobid:${latest_grobid_version}-crf
```
